server/configuration.go

```diff
@@ -70,14 +73,23 @@ func (p *Plugin) setConfiguration(configuration *configuration) {
 
 // OnConfigurationChange is invoked when configuration changes may have been made.
 func (p *Plugin) OnConfigurationChange() error {
-	var configuration = new(configuration)
+	oldConfig := p.getConfiguration()
+	newConfig := new(configuration)
 
 	// Load the public configuration fields from the Mattermost server configuration.
-	if err := p.API.LoadPluginConfiguration(configuration); err != nil {
+	if err := p.API.LoadPluginConfiguration(newConfig); err != nil {
 		return errors.Wrap(err, "failed to load plugin configuration")
 	}
 
-	p.setConfiguration(configuration)
+	p.setConfiguration(newConfig)
+
+	if oldConfig.TheSetting != newConfig.TheSetting {
+		// in case we want to include the config values in the websocket message itself
+		payload := map[string]interface{}{
+			"the_setting": newConfig.TheSetting,
```

Comment:
This should be `TheSetting` instead of `the_setting` to be consistent with the config's field